packages/core/src/locales/fr.ts

```diff
@@ -0,0 +1,132 @@
+import type { $ZodStringFormats } from "../checks.js";
+import type * as errors from "../errors.js";
+import * as util from "../util.js";
+
+const Sizable: Record<string, { unit: string; verb: string }> = {
+  string: { unit: "caractères", verb: "avoir" },
+  file: { unit: "octets", verb: "avoir" },
+  array: { unit: "éléments", verb: "avoir" },
+  set: { unit: "éléments", verb: "avoir" },
+};
+
+function getSizing(origin: string): { unit: string; verb: string } | null {
+  return Sizable[origin] ?? null;
+}
+
+export const parsedType = (data: any): string => {
+  const t = typeof data;
+
+  switch (t) {
+    case "number": {
+      return Number.isNaN(data) ? "NaN" : "nombre";
+    }
+    case "object": {
+      if (Array.isArray(data)) {
+        return "tableau";
+      }
+      if (data === null) {
+        return "null";
+      }
+
+      if (
+        Object.getPrototypeOf(data) !== Object.prototype &&
+        data.constructor
+      ) {
+        return data.constructor.name;
+      }
+    }
+  }
+  return t;
+};
+
+const Nouns: {
+  [k in $ZodStringFormats | (string & {})]?: string;
+} = {
+  regex: "entrée",
+  email: "adresse e-mail",
+  url: "URL",
+  emoji: "emoji",
+  uuid: "UUID",
+  uuidv4: "UUIDv4",
+  uuidv6: "UUIDv6",
+  nanoid: "nanoid",
+  guid: "GUID",
+  cuid: "cuid",
+  cuid2: "cuid2",
+  ulid: "ULID",
+  xid: "XID",
+  ksuid: "KSUID",
+  datetime: "date et heure ISO",
+  date: "date ISO",
+  time: "heure ISO",
+  duration: "durée ISO",
+  ipv4: "adresse IPv4",
+  ipv6: "adresse IPv6",
+  cidrv4: "plage IPv4",
+  cidrv6: "plage IPv6",
+  base64: "chaîne encodée en base64",
+  base64url: "chaîne encodée en base64url",
+  json_string: "chaîne JSON",
+  e164: "numéro E.164",
+  jwt: "JWT",
+  template_literal: "entrée",
+};
+
+const error: errors.$ZodErrorMap = (issue) => {
+  switch (issue.code) {
+    case "invalid_type":
+      return `Entrée invalide : ${issue.expected} attendu, ${parsedType(issue.input)} reçu`;
+    case "invalid_value":
+      if (issue.values.length === 1)
+        return `Entrée invalide : ${util.stringifyPrimitive(issue.values[0])} attendu`;
+      return `Option invalide : une valeur parmi ${util.joinValues(issue.values, "|")} attendue`;
+    case "too_big": {
+      const adj = issue.inclusive ? "<=" : "<";
+      const sizing = getSizing(issue.origin);
+      if (sizing)
+        return `Trop grand : ${issue.origin ?? "valeur"} doit ${sizing.verb} ${adj}${issue.maximum.toString()} ${sizing.unit ?? "éléments"}`;
```

Comment:
I'm not sure if there is a way to handle pluralization or not. If not, maybe `élément(s)` would be better. Mainly to avoid incorrect phrasing when the max value is 1.